examples/shader/shader_defs.rs

```diff
@@ -87,15 +87,16 @@ struct CustomMaterial {
 // In this case, we specialize on whether or not to configure the "IS_RED" shader def.
 // Specialization keys should be kept as small / cheap to hash as possible,
 // as they will be used to look up the pipeline for each drawn entity with this material type.
```

Comment:
Maybe update comment here to explain this has to be Pod repr(C) and why if possible